src/ripple/app/ledger/InboundLedger.cpp

```diff
@@ -568,6 +577,12 @@ void InboundLedger::trigger (Peer::ptr const& peer)
                         {
                             * (tmGL.add_nodeids ()) = id.getRawString ();
                         }
+
+                        // If we're not querying for a lot of entries,
+                        // query extra deep
+                        if (nodeIDs.size() <= fetchSmallNodes)
+                            tmGL.set_querydepth (tmGL.querydepth() + 1);
```

Comment:
FYI, another way to write this is `*tmGL.mutable_querydepth() += 1;`  It's a tiny bit more efficient in some cases because of one less method call, but here the compiler will almost certainly optimize both calls out.